dttools/src/jx_eval.c

```diff
@@ -186,48 +186,43 @@ static struct jx *jx_eval_array(struct jx_operator *op, struct jx *left, struct
 static struct jx *jx_eval_call(struct jx *func, struct jx *args, struct jx *ctx) {
 	assert(func);
 	assert(args);
-	assert(args->type == JX_ARRAY);
-
-	if (!jx_istype(func, JX_SYMBOL)) {
-		jx_error(jx_format(
-			"on line %d, unknown function: %s",
-			func->line,
-			func->u.symbol_name
-		));
-	}
+	assert(jx_istype(args, JX_ARRAY));
+	assert(jx_istype(func, JX_SYMBOL)); //TODO
```

Comment:
That assert seems correct to me, why TODO?

Reply:
Whoops, forgot that one! I wanted to check if the parser only accepts symbols there (so an assert is correct) or if it's possible to get other types (so an error message would be correct)

Reply:
Aha!

    expression: "bad_call"()
    jx_eval_call: jx_eval.c:190[DEVELOPMENT]: Assertion 'jx_istype(func, JX_SYMBOL)' failed.
    Aborted (core dumped)

Reply:
Good catch.  I would be satisfied if this failed at parsing rather than execution.  Unless you can think of a reason for having expressions that evaluate to functions?

Reply:
Not really. We currently don't eval function names, so you can't do higher-order things